.github/workflows/ci.yml

```diff
@@ -92,72 +117,76 @@ jobs:
             cd build
             cmake -A x64 -DCMAKE_TOOLCHAIN_FILE=${VCPKG_INSTALLATION_ROOT}/scripts/buildsystems/vcpkg.cmake \
-                         -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install \
+                         -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install/deps \
                          -DCMAKE_BUILD_TPYE=${{matrix.build_type}} \
                          -DIDYNTREE_USES_ICUB_MAIN:BOOL=ON \
-                         -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
+                         -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install/deps ..
             cmake --build . --config ${{matrix.build_type}} --target INSTALL
                            
              
         - name: Source-based Dependencies [Ubuntu/macOS]
-          if: matrix.os == 'ubuntu-latest' || matrix.os == 'macOS-latest'
+          if: steps.cache-source-deps.outputs.cache-hit != 'true' && (matrix.os == 'ubuntu-latest' || matrix.os == 'macOS-latest')
           shell: bash
           run: |
             # YCM
             cd ${GITHUB_WORKSPACE}
-            git clone https://github.com/robotology/ycm
+            git clone -b ${YCM_TAG} https://github.com/robotology/ycm
             cd ycm
             mkdir -p build
             cd build
-            cmake -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
+            cmake -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install/deps ..
             cmake --build . --config ${{matrix.build_type}} --target install
              
             # YARP
             cd ${GITHUB_WORKSPACE}
             git clone https://github.com/robotology/yarp
             cd yarp
+            git checkout ${YARP_TAG}
             mkdir -p build
             cd build
-            cmake -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install \
-                  -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
+            cmake -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install/deps \
+                  -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install/deps ..
             cmake --build . --config ${{matrix.build_type}} --target install
                                                     
             # ICUB
             cd ${GITHUB_WORKSPACE}
             git clone https://github.com/robotology/icub-main
             cd icub-main
+            git checkout ${ICUB_TAG}
             mkdir -p build
             cd build
-            cmake -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install \
-                  -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
+            cmake -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install/deps \
+                  -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install/deps ..
             cmake --build . --config ${{matrix.build_type}} --target install
             
             # iDynTree
             cd ${GITHUB_WORKSPACE}
             git clone https://github.com/robotology/iDynTree
             cd iDynTree
+            git checkout ${iDynTree_TAG}
             git checkout devel
```

Comment:
Probably you forgot to remove this line.

Reply:
I removed this line in the recent commits https://github.com/robotology/whole-body-estimators/commit/0404ba1760e720f3d40a0c74d56f8bce0ec38f5f